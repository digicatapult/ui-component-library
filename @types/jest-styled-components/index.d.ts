declare module 'jest-styled-components/serializer'
