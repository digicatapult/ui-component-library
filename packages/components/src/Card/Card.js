import React from 'react'
import { PropTypes } from 'prop-types'
import styled from 'styled-components'

import colors from '../colors'

const { black, grey } = colors

const Text = styled.div`
  font-size: 14px;
`

const Title = styled.h2`
  font-size: 16px;
  margin: 0 0 15px 0;
`

const Description = styled.div`
  font-size: 14px;
  padding-top: 1rem;

  -webkit-line-clamp: 8;
  -webkit-box-orient: vertical;
  overflow: hidden;
  text-overflow: ellipsis;
  display: -webkit-box;
`

const Container = styled.div`
  display: flex;
  flex-flow: column nowrap;

  margin: 2rem;
  &:hover {
    box-shadow: 10px 10px 4px rgba(0, 0, 0, 0.25);
  }

  padding: 10px;
  margin-bottom: 1rem;

  border: 2px solid ${grey};
  border-radius: 5px;
  box-sizing: border-box;

  width: 240px;
  height: 240px;
  cursor: pointer;
`

const Card = ({ title, name, date, description, ...rest }) => {
  return (
    <Container {...rest}>
      <Title>{title}</Title>
      <Text>by {name}</Text>
      <Text>
        on <strong>{new Date(date).toLocaleDateString()}</strong>
      </Text>
      <Description>{description?.slice(0, 100).trim()}...</Description>
    </Container>
  )
}

Card.propTypes = {
  title: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  date: PropTypes.instanceOf(Date).isRequired,
  description: PropTypes.string.isRequired,
}

export default Card
