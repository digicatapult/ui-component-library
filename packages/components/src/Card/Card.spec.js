import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import Card from '.'

describe('Card', () => {
  test('default', () => {
    const tree = renderer
      .create(
        <Card
          data={{
            title: 'Card story',
            editors: [{ firstName: 'Story Writer' }],
            start: 1613384947838,
            description:
              'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Est ullamcorper eget nulla facilisi etiam dignissim diam quis. Sem et tortor consequat id porta. Viverra nam libero justo laoreet sit.'
          }}
          handleClick={() => {}}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
