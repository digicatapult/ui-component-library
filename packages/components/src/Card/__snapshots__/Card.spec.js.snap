// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card default 1`] = `
.c2 {
  font-size: 14px;
}

.c1 {
  font-size: 16px;
  margin: 0 0 15px 0;
}

.c3 {
  font-size: 14px;
  padding-top: 1rem;
  -webkit-line-clamp: 8;
  -webkit-box-orient: vertical;
  overflow: hidden;
  text-overflow: ellipsis;
  display: -webkit-box;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  margin: 2rem;
  padding: 10px;
  margin-bottom: 1rem;
  border: 1px solid #000000;
  box-sizing: border-box;
  width: 240px;
  height: 240px;
  cursor: pointer;
}

.c0:hover {
  box-shadow: 10px 10px 4px rgba(0,0,0,0.25);
}

<div
  className="c0"
  onClick={[Function]}
>
  <h2
    className="c1"
  >
    Card story
  </h2>
  <div
    className="c2"
  >
    by 
    Story Writer
  </div>
  <div
    className="c2"
  >
    on 
    <strong>
      2/15/2021
    </strong>
  </div>
  <div
    className="c3"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore
    ...
  </div>
</div>
`;
